### PR TITLE
Remove py2 things from lib

### PR DESCRIPTION
**Pull request by IAlibay**
Towards #2541

Changes made in this Pull Request:
 - Removes six & __future__ imports from lib

Question:
 - <s>Do we want changelog entries for these things?</s>

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/lib/util.py [hunk 1]
```diff
@@ -438,7 +435,7 @@ def _get_stream(filename, openfunction=open, mode='r'):
         # case we have to ignore the error and return None. Second is when openfunction can't open the file because
         # either the file isn't there or the permissions don't allow access.
         if errno.errorcode[err.errno] in ['ENOENT', 'EACCES']:
-            six.reraise(*sys.exc_info())
+            raise sys.exc_info()[0](sys.exc_info()[1]) from err
```
Comment (IAlibay) [inline package/MDAnalysis/lib/util.py:438]: I'm not sure this is the right conversion here, any input would be appreciated

**Comment by IAlibay** [comment]
Reverted to raising from `None` in c23b255, after re-reading #2357 it seemed like this was the behaviour we wanted. Left in the "except *Error as exc" for now, but if we go for `None` I'm not sure we actually need it.

**Comment by lilyminium** [comment]
I didn't update CHANGELOG in #2750. I guess if we did it would go in Deprecations?

**Comment by IAlibay** [comment]
> I didn't update CHANGELOG in #2750. I guess if we did it would go in Deprecations?

Spoke to @richardjgowers off-github, and his take on this was that changelog changes for this won't be necessary (my bad, should have removed my question yesterday).

File: package/MDAnalysis/lib/distances.py [hunk 2]
```diff
@@ -99,12 +95,10 @@ def _run(funcname, args=None, kwargs=None, backend="serial"):
     backend = backend.lower()
     try:
         func = getattr(_distances[backend], funcname)
-    except KeyError:
-        raise_from(
-            ValueError(
-                "Function {0} not available with backend {1}; try one "
-                "of: {2}".format(funcname, backend, _distances.keys())),
-            None)
+    except KeyError as exc:
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/lib/distances.py]: Fixed in 6b9a257

File: package/MDAnalysis/lib/util.py [hunk 3]
```diff
@@ -179,10 +179,8 @@
    underlying stream and ``NamedStream.close(force=True)`` will also
    close it.
 """
-from __future__ import division, absolute_import
-import six
-from six.moves import range, map
 import sys
+import six
```
Comment (lilyminium) [inline package/MDAnalysis/lib/util.py:183]:
```suggestion
```
missed one :-)

**Pull request merged**